### PR TITLE
fix cannot read data when opening a deeplink

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
The initial error was caused by the launch parsing of deep links where the deeplink did not match the patterns for channel or permalink, so in this PR apart of fixing the issue when the deep link is invalid we also handle a deeplink for a server that we do not have registered yet by presenting the server screen an populating with the url of the deep link

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-58467

#### Checklist
- [x] Added or updated unit tests (required for all new features)
#### Release Note

```release-note
Fixed deep links that did not match the patterns of channel or permalink and was crashing the app
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]

**Review by rahimrahman — APPROVED** [review]
I have only one extra test case I thought could be useful, but not a blocker.

File: app/utils/deep_link/index.test.ts [hunk 1]
```diff
@@ -208,6 +239,23 @@ describe('getLaunchPropsFromDeepLink', () => {
...
+            url: 'https://existingserver.com/login',
+        };
+        const result = getLaunchPropsFromDeepLink('https://existingserver.com/login', true);
+
+        expect(result).toEqual({
+            launchType: Launch.DeepLink,
+            coldStart: true,
+            extra: extraData,
+        });
+    });
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/utils/deep_link/index.test.ts:269]: done

**Comment by saturninoabril** [comment]
Requesting review from Lindy for she was able to repro the issue - https://github.com/mattermost/mattermost/issues/26803#issuecomment-2137994652

For the failed Detox (on build), it's not related here and was happening even from main. Will investigate separately - https://community.mattermost.com/core/pl/t1zi5wwic3fdidzeh51z67gybc

**Comment by lindy65** [comment]
@enahum - building apps for the PR is [failing](https://community.mattermost.com/core/pl/zxf7ituhfbf1dqdgkniesn78ew)

**Comment by enahum** [comment]
@lindy65 build successful https://community.mattermost.com/core/pl/ermnggnht3rbzbqhauaqxbzo4r

**Review by lindy65 — APPROVED** [review]
Thanks @enahum - I have tested according to repro steps and I am taken to the account creation page in a browser. Creating an account works without issue and I can log into MM mobile browser

**Pull request merged**